components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,76 @@
+const reddit = require("../../reddit.app.js");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emits an event each time a new link is added to a subreddit",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "The subreddit you'd like to watch for new links.",
+      default: "redditdev",
+    },
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for new links on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minutes.
+      },
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emits 10 sample events on the first run during deploy.
+      let before = null;
+
+      const reddit_things = await this.reddit.getNewSubredditLinks(
+        null,
+        this.subreddit,
+        10
+      );
+
+      const links_pulled = this.reddit.wereLinksPulled(reddit_things);
+      if (links_pulled) {
+        before = reddit_things.data.children[0].data.name;
+        reddit_things.data.children.forEach((reddit_link) => {
```

Comment:
When I tested, I realized these events are emitted in reverse chronological order (most recent link is emitted first, so it ends up at the bottom of the list of sample links). Could you [reverse the array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse) before emitting, so events are always ordered with the most recent link at the top?

Reply:
Yeah sure. I applied the resverse and noticed the order is as you point out.
I will check on other sources, that it is preferred most recent event is emitted last.